File: frontend/src/api/backtest.ts
import { request } from "./client";
import type { BacktestRequest, BacktestResponse } from "../types/backtest";

/**
 * 调用 POST /api/backtest。
 * 回测需先拉取行情再向量计算，可能耗时较长，故给 60s 超时（client 默认仅 15s）。
 */
export function runBacktest(req: BacktestRequest): Promise<BacktestResponse> {
  return request<BacktestResponse>("/api/backtest", {
    method: "POST",
    body: req,
    timeoutMs: 60_000,
  });
}

File: backend/app/api/backtest.py
"""POST /api/backtest — thin router that delegates to the backtest service."""

from __future__ import annotations

from fastapi import APIRouter

from app.backtest.schemas import BacktestRequest, BacktestResponse
from app.backtest.service import run_backtest

router = APIRouter(prefix="/api", tags=["backtest"])


@router.post("/backtest")
async def backtest(req: BacktestRequest) -> BacktestResponse:
    return await run_backtest(req)
